src/POData/Providers/Metadata/ResourceFunctionType.php

```diff
@@ -18,12 +18,15 @@ class ResourceFunctionType
      */
     private $baseType = null;
 
+    private $resourceType = null;
+
     /**
      * ResourceFunctionType constructor.
      * @param string|array $functionName
      * @param FunctionImportAnonymousType $type
+     * @param ResourceType $resource
      */
-    public function __construct($functionName, FunctionImportAnonymousType $type)
+    public function __construct($functionName, FunctionImportAnonymousType $type, ResourceType $resource)
```

Comment:
Need for this revealed by integration testing - ProvidersWrapper lost plot with null ref exception